worker/codalabworker/aws_batch.py

```diff
@@ -0,0 +1,539 @@
+            batch_client=self._batch_client,
+            queue_name=queue_name,
+            worker=self._worker,
+            bundle=bundle,
+            bundle_path=bundle_path,
+            resources=resources
+        )
+        return run
+
+
+class AwsBatchRun(FilesystemRunMixin, RunBase):
+    """
+    This class manages a single run on AWS Batch.
+    This is similar to the standard Run, but all execution happens in a Batch job rather than locally on the worker.
+
+    It also has the added constraint for the shared file system option, that the file system must be shared properly
+    on the compute cluster which is serving the Batch queue.
+    For an article on how to achieve this, see:
+    """
+    def __init__(self, bundle_service, batch_client, queue_name, worker, bundle, bundle_path, resources):
+        super(AwsBatchRun, self).__init__()
+        self._bundle_service = bundle_service
+        self._batch_client = batch_client
+        self._queue_name = queue_name
+        self._worker = worker
+        self._bundle = bundle
+        self._uuid = bundle['uuid']
+        self._bundle_path = bundle_path
+        self._resources = resources
+        self._fsm = None
+
+    @property
+    def is_shared_file_system(self):
+        return self._worker.shared_file_system
+
+    @property
+    def bundle(self):
+        return self._bundle
+
+    @property
+    def resources(self):
+        return self._resources
+
+    @property
+    def bundle_path(self):
+        return self._bundle_path
+
+    def start(self):
+        # Report that the bundle is running. We note the start time here for
+        # accurate accounting of time used, since the clock on the bundle
+        # service and on the worker could be different.
+        start_message = {
+            'hostname': socket.gethostname(),
+            'start_time': current_time(),
+        }
+        if not self._bundle_service.start_bundle(self._worker.id, self._uuid, start_message):
+            return False
+
+        if self.is_shared_file_system:
+            # On a shared file system we create the path in the bundle manager
+            # to avoid NFS directory cache issues. Here, we wait for the cache
+            # on this machine to expire and for the path to appear.
+            while not os.path.exists(self._bundle_path):
+                time.sleep(0.5)
+        else:
+            # Set up a directory to store the bundle.
+            remove_path(self._bundle_path)
+            os.mkdir(self._bundle_path)
+
+        self.create_fsm().start()
+        return True
+
+    def resume(self):
+        """
+        Report that the bundle is running. We note the start time here for
+        accurate accounting of time used, since the clock on the bundle
+        service and on the worker could be different.
+        """
+        start_message = {
+            'hostname': socket.gethostname(),
+            'start_time': current_time(),
```

Comment:
I actually don't think it makes sense here since the `time` a bundle runs for batch is calculated via the batch job not via the time the worker 'started'. I'm just going to remove this.